internal/server/placein/placein.go

```diff
@@ -40,38 +42,77 @@ func GetPlacesIn(
 	if err := util.CheckValidDCIDs(parentPlaces); err != nil {
 		return nil, err
 	}
-
-	// Place relations are from base geo imports. Only trust the base cache.
-	btDataList, err := bigtable.Read(
-		ctx,
-		store.BtGroup,
-		bigtable.BtPlacesInPrefix,
-		[][]string{parentPlaces, {childPlaceType}},
-		func(jsonRaw []byte) (interface{}, error) {
-			var containedInPlaces pb.ContainedPlaces
-			err := proto.Unmarshal(jsonRaw, &containedInPlaces)
-			return containedInPlaces.Dcids, err
-		},
-	)
-	if err != nil {
-		return nil, err
-	}
 	result := map[string][]string{}
-	processed := map[string]struct{}{}
-	for _, parent := range parentPlaces {
-		if _, ok := processed[parent]; ok {
-			continue
+
+	if store.BtGroup != nil {
+		// Place relations are from base geo imports. Only trust the base cache.
+		btDataList, err := bigtable.Read(
+			ctx,
+			store.BtGroup,
+			bigtable.BtPlacesInPrefix,
+			[][]string{parentPlaces, {childPlaceType}},
+			func(jsonRaw []byte) (interface{}, error) {
+				var containedInPlaces pb.ContainedPlaces
+				err := proto.Unmarshal(jsonRaw, &containedInPlaces)
+				return containedInPlaces.Dcids, err
+			},
+		)
+		if err != nil {
+			return nil, err
 		}
-		// Go through (ordered) import groups one by one, stop when data is found.
-		for _, btData := range btDataList {
-			for _, row := range btData {
-				if row.Parts[0] == parent {
-					result[parent] = row.Data.([]string)
-					processed[parent] = struct{}{}
-					break
+		processed := map[string]struct{}{}
+		for _, parent := range parentPlaces {
+			if _, ok := processed[parent]; ok {
+				continue
+			}
+			// Go through (ordered) import groups one by one, stop when data is found.
+			for _, btData := range btDataList {
+				for _, row := range btData {
+					if row.Parts[0] == parent {
+						result[parent] = row.Data.([]string)
+						processed[parent] = struct{}{}
+						break
+					}
 				}
 			}
 		}
 	}
+
+	if store.SQLiteClient != nil {
+		// Only queries based on direct containedInPlace for now.
+		// This could extend to more hops and even link with BT cache data, but that
+		// might make it too complicated.
+		// In custom DC, it's reasonable to ask user to provide direct containment
+		// relation.
+		parentPlacesStr := "'" + strings.Join(parentPlaces, "', '") + "'"
+		query := fmt.Sprintf(
```

Comment:
Do we care about SQL injection here?
Since this is open-source code, maybe we want to avoid SQL injection?
https://go.dev/doc/database/sql-injection
@pradh

Reply:
Totally. I updated here and a few other places to use query args in stead of raw strings.